content/en/docs/gke/deploy/upgrade.md

```diff
@@ -24,71 +38,167 @@ However, specific upgrades might need manual actions below.
+   # Verify the context connects to the cluster properly
+   kubectl get namespace
+   ```
+
+   If you are using a different enviroment, you can always
+   reconfigure the context by:
+
+   ```bash
+   make create-context
+   ```
+
+1. Check your existing config connector version:
+   ```bash
+   # For Kubeflow v1.1, it should be 1.15.1
+   $ kubectl get namespace cnrm-system -ojsonpath='{.metadata.annotations.cnrm\.cloud\.google\.com\/version}'
+   1.15.1
+   ```
+1. Uninstall the old config connector in the management cluster:
+   ```bash
+   kubectl delete sts,deploy,po,svc,roles,clusterroles,clusterrolebindings --all-namespaces -l cnrm.cloud.google.com/system=true --wait=true
+   kubectl delete validatingwebhookconfiguration abandon-on-uninstall.cnrm.cloud.google.com --ignore-not-found --wait=true
+   kubectl delete validatingwebhookconfiguration validating-webhook.cnrm.cloud.google.com --ignore-not-found --wait=true
+   kubectl delete mutatingwebhookconfiguration mutating-webhook.cnrm.cloud.google.com --ignore-not-found --wait=true
+   ```
+   These commands uninstall the config connector without removing your resources.
+1. Replace your `./Makefile` with the version in Kubeflow `v1.2.0`: https://github.com/kubeflow/gcp-blueprints/blob/v1.2.0/management/Makefile.
+
+   If you made any customizations in `./Makefile`, you should merge your changes with the upstream version. We've refactored the Makefile to move substantial commands into the upstream package, so hopefully future upgrades won't require a manual merge of the Makefile.
+
+1. Update `./upstream/management` package:
+   ```bash
+   make update
+   ```
+1. Use kpt to set user values:
+   ```bash
+   kpt cfg set -R . name ${NAME}
+   kpt cfg set -R . gcloud.core.project ${PROJECT}
+   kpt cfg set -R . location ${LOCATION}
+   ```
+   Note, you can find out which setters exist in a package and what there current values are by:
+   ```bash
+   kpt cfg list-setters .
+   ```
+1. Apply upgraded config connector:
+   ```bash
+   make apply-kcc
+   ```
+   Note, you can optionally also run `make apply-cluster`, but it should be the same as your existing management cluster.
+1. Check your config connector upgrade is successful:
+   ```bash
+   # For Kubeflow v1.2, it should be 1.29.0
+   $ kubectl get namespace cnrm-system -ojsonpath='{.metadata.annotations.cnrm\.cloud\.google\.com\/version}'
+   1.29.0
+   ```
+
+## Upgrading Kubeflow cluster
+
+**DISCLAIMERS**:
+
+- The upgrade process depends on each Kubeflow application to handle the upgrade properly. There's no guarantee on data completeness unless the application provides such a guarantee.
+- Always recommend backing up your data before an upgrade.
+- Upgrading Kubeflow cluster can be a disruptive process, please schedule some downtime and communicate with your users.
+
+For upgrading from specific versions, there can be manual actions needed, so recommend referring to the specific sections below.
```

Comment:
Toward an active voice , e.g.

```suggestion
To upgrade from specific versions of Kubeflow, you may need to take certain manual actions—refer to specific sections in the guidelines below.
```

WDYT?

Reply:
This is great, thank you!